cylc/flow/task_pool.py

```diff
@@ -895,7 +895,7 @@ def can_stop(self, stop_mode):
         return not any(
             (
                 stop_mode in [StopMode.REQUEST_CLEAN, StopMode.REQUEST_KILL]
-                and itask.state(*TASK_STATUSES_ACTIVE)
+                and itask.state(*TASK_STATUSES_ACTIVE, TASK_STATUS_PREPARING)
```

Comment:
Not sure about this one.

Reply:
OK as-is I think. We can safely shut down with (unsubmitted) `preparing` tasks, because they will be re-prepped on restart as noted above.

Reply:
Makes sense, reverted the change, added a comment and an integration test.